### DISCUSSION TITLE
ADR Suggestion, Generic Theory Class

### DISCUSSION THREAD

**Discussion by rozyczko** [easyscience/.github#5] [Ideas]
Another discussion https://github.com/EasyScience/EasyDiffractionLib/discussions/97 goes into design details of the `Phase` class, used to manipulate the atomistic object, with properties like `atoms`, `cell`, `background` etc.
This is a good picture for diffraction and spectroscopy (and Bragg Edge), where the underlying model is a system composed of atoms.
However, for techniques like reflectometry, this isn't the case. Here, the system is a set of layers and surfaces.
For QENS this is even more simple, with the model being a set of simple functions.

This means the base class for this descriptor should be a bit simpler than the `Phase` we discussed.
For starters, it can't be called `Phase`. How about calling this a `Theory`? Or specifically `BaseTheory`, considering those are models we use to generate the theory curve to compare/fit to the experiment.

Diffraction would use it as a base class for `Phase`, refl for `Layer` etc.

**Comment by AndrewSazonov** [comment]
Why not just use `model`? `Model` and `BaseModel` seem more natural to me. And, regardless of the technique, there is always a physical model, which can be a `phase`, `crystal`, `layer`, etc.

**Comment by rozyczko** [comment]
I guess because `model` is already a very loaded term.
We use "model" to describe multiple objects - Qt models, data models for calculators, fit models etc.

Adding another concept of `model` would make it more confusing, I think.

**Comment by AndrewSazonov** [comment]
I agree that there are already a number things called `model` in our projects, and it can be confusing. But if I saw an object called `theory`, it would confuse me even more. And at the moment I personally don't see a better name than `model`.  Maybe there are other ideas besides `model` and `theory`?

**Comment by rozyczko** [comment]
But it **is** a theory, right? We have experimental data to which we fit our theoretical description of the system.
The profile we get from the atomistic model is a theoretical profile. So just like for experimental data and parameters we have `Experiment`, for theoretical data and parameters we should have `Theory`.
Again, this is about comparing _theory_ to _experiment_.

**Comment by rozyczko** [comment]
`TheoreticalModel`

Grudgingly accepted consensus.

For technique-specific implementations we would derive from `TheoreticalModel` to create say
`Sample` or `Phase` for diffraction
`Layer` for reflectometry
etc

**Comment by rozyczko** [comment]
... and also covered in more detail and depth in https://github.com/orgs/easyscience/discussions/29

Therefore, closing this discussion.
